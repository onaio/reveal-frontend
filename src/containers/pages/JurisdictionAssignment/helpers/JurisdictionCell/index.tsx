/** component in the jurisdictionTable selector that will render a node according to a few rules:
 * 1) as a link if the node has children
 * 2) as plain text if the node does not have children.
 */
import React from 'react';
import { Link } from 'react-router-dom';
import { TreeNode } from '../../../../../store/ducks/opensrp/hierarchies/types';

/**
 * Props for NodeCell
 */
export interface NodeCellProps {
  node: TreeNode /** the current jurisdiction */;
  baseUrl: string;
}

/**
 * NodeCell
 *
 * Displays the jurisdiction.  If the jurisdiction is not a leaf node, then it is
 * displayed as a link, so that one can traverse down the tree.
 * @param props - the props
 */
const NodeCell = (props: NodeCellProps) => {
  const { node, baseUrl } = props;

  // isLeafNode if node does not have children
  const isLeafNode = !node.hasChildren();

  if (isLeafNode) {
    return <span key={`${node.model.id}-span`}>{node.model.label}</span>;
  }

  return (
    <Link to={`${baseUrl}/${node.model.id}`} key={`${node.model.id}-span`}>
      {node.model.label}
    </Link>
  );
};

export { NodeCell };
