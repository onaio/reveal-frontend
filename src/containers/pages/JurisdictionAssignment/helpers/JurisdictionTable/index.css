/* allows tooltip to show on disabled button */
.btn[disabled] {
  pointer-events: none;
}
