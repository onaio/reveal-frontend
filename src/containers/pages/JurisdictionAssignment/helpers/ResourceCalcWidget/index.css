/* Chrome, Safari, Edge, Opera */

#resource-calc-widget input::-webkit-outer-spin-button,
#resource-calc-widget input::-webkit-inner-spin-button {
  -webkit-appearance: none;
  margin: 0;
}

/* Firefox */
#resource-calc-widget input[type='number'] {
  -moz-appearance: textfield;
}

#resource-calc-widget input {
  width: 45px;
}
