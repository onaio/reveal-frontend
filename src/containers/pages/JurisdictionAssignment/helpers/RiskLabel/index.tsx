/** Rendered as a table cell and shows the risk label for a given jurisdiction node */
import React from 'react';
import { TreeNode } from '../../../../../store/ducks/opensrp/hierarchies/types';
import { JurisdictionsMetadata } from '../../../../../store/ducks/opensrp/jurisdictionsMetadata';

export interface RiskLabelProps {
  metadata: JurisdictionsMetadata[];
  node: TreeNode;
}

/** finds the nodes risk label from the fetched metadata and displays it */
export const RiskLabel = (props: RiskLabelProps) => {
  const { metadata, node } = props;

  // could get more that one meta objects
  const metasOfInterest = metadata.filter(meta => meta.key === node.model.id);

  // pick the first one with a value
  const metaOfInterest = metasOfInterest.filter(meta => meta.value !== undefined);
  const theSingleMeta = metaOfInterest[0];

  if (theSingleMeta && theSingleMeta.value) {
    return <>{theSingleMeta.value}</>;
  }
  return <>{'--'}</>;
};
