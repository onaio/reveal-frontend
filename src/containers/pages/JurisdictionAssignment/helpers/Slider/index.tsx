/** renders the range slider component that is used to set the
 * threshold from which we can filter out the jurisdictions to be
 * auto-targeted based on their risk value
 */
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import reducerRegistry from '@onaio/redux-reducer-registry';
import React, { useState } from 'react';
import InputRange, { Range } from 'react-input-range';
import 'react-input-range/lib/css/index.css';
import { connect } from 'react-redux';
import { Button } from 'reactstrap';
import Col from 'reactstrap/lib/Col';
import Row from 'reactstrap/lib/Row';
import { ActionCreator, Store } from 'redux';
import {
  ADJUST_SLIDER_MESSAGE,
  CONTINUE_TO_NEXT_STEP,
  NUMBER_OF_STRUCTURES_IN_JURISDICTIONS,
} from '../../../../../configs/lang';
import { PlanDefinition } from '../../../../../configs/settings';
import { RISK_LABEL } from '../../../../../constants';
import hierarchyReducer, {
  autoSelectNodes,
  fetchTree,
  getStructuresCount,
  getTreeById,
  reducerName as hierarchyReducerName,
} from '../../../../../store/ducks/opensrp/hierarchies';
import { SELECTION_REASON } from '../../../../../store/ducks/opensrp/hierarchies/constants';
import { TreeNode } from '../../../../../store/ducks/opensrp/hierarchies/types';
import jurisdictionMetadataReducer, {
  fetchJurisdictionsMetadata,
  FetchJurisdictionsMetadataAction,
  getJurisdictionsMetadata,
  JurisdictionsMetadata,
  reducerName,
} from '../../../../../store/ducks/opensrp/jurisdictionsMetadata';
import './slider.css';

reducerRegistry.register(reducerName, jurisdictionMetadataReducer);
reducerRegistry.register(hierarchyReducerName, hierarchyReducer);

/** props for JurisdictionSelectionSlider */
interface Props {
  rootJurisdictionId: string;
  tree?: TreeNode;
  structuresCount: number;
  fetchJurisdictionsMetadataCreator: ActionCreator<FetchJurisdictionsMetadataAction>;
  jurisdictionsMetadata: JurisdictionsMetadata[];
  autoSelectCreator: typeof autoSelectNodes;
  fetchTreeCreator: typeof fetchTree;
  plan: PlanDefinition;
  onClickNext: () => void;
}

const defaultProps = {
  autoSelectCreator: autoSelectNodes,
  fetchJurisdictionsMetadataCreator: fetchJurisdictionsMetadata,
  fetchTreeCreator: fetchTree,
  jurisdictionsMetadata: [],
  onClickNext: () => {
    return;
  },
  rootJurisdictionId: '',
  structuresCount: 0,
};

/** renders a draggable slider that sets a value used as the
 * threshold for autoSelecting jurisdictions.
 */
export const JurisdictionSelectionsSlider = (props: Props) => {
  const [value, setValue] = useState<number | Range>(0);

  const {
    rootJurisdictionId,
    structuresCount,
    autoSelectCreator,
    jurisdictionsMetadata,
    plan,
    tree,
  } = props;

  const planId = plan.identifier;

  const onChangeComplete = (val: number | Range) => {
    const metaJurOfInterest = jurisdictionsMetadata.filter(
      metaObject => parseInt(metaObject.value, 10) > val
    );
    const jurisdictionsIdsMeta = metaJurOfInterest.map(meta => meta.key);
    const callback = (node: TreeNode) => {
      const isLeafNodePastThreshHold =
        !node.hasChildren() && jurisdictionsIdsMeta.includes(node.model.id);
      return isLeafNodePastThreshHold;
    };
    autoSelectCreator(rootJurisdictionId, callback, planId, SELECTION_REASON.AUTO_SELECTION);
  };

  React.useEffect(() => {
    onChangeComplete(value);
  }, []);

  const onChange = (val: number | Range) => setValue(val);

  if (!plan || !tree) {
    return null;
  }

  return (
    <div className="mt-3">
      <Row>
        <Col xs="12" md={{ size: 8, offset: 2 }} style={{ textAlign: 'center' }}>
          <h4 className="mb-5 font-weight-bold">{ADJUST_SLIDER_MESSAGE}</h4>
          <Row className="auto-target-row mb-3">
            <Col xs="12" md={6} className="slider-section py-3 px-5">
              <p>
                {RISK_LABEL}&nbsp;&nbsp;<span className="risk-label">{`${value}%`}</span>
              </p>
              <InputRange
                maxValue={100}
                minValue={0}
                value={value}
                onChange={onChange}
                onChangeComplete={onChangeComplete}
              />
            </Col>
            <Col xs="12" md={6} className="info-section p-3">
              <p>
                <FontAwesomeIcon icon="home" />
              </p>
              <p>{NUMBER_OF_STRUCTURES_IN_JURISDICTIONS}</p>
              <p>{structuresCount}</p>
            </Col>
          </Row>
        </Col>
      </Row>
      <hr />
      <Button className="btn btn-success float-right mt-3" onClick={props.onClickNext}>
        {CONTINUE_TO_NEXT_STEP}
      </Button>
    </div>
  );
};

JurisdictionSelectionsSlider.defaultProps = defaultProps;

/** map state to props */
export type MapStateToProps = Pick<Props, 'structuresCount' | 'tree' | 'jurisdictionsMetadata'>;
/** map dispatch to action creators */
export type MapDispatchToProps = Pick<
  Props,
  'autoSelectCreator' | 'fetchJurisdictionsMetadataCreator' | 'fetchTreeCreator'
>;

const structureCountSelector = getStructuresCount();
const treeSelector = getTreeById();

const mapStateToProps = (state: Partial<Store>, ownProps: Props): MapStateToProps => {
  const filters = {
    planId: ownProps.plan.identifier,
    rootJurisdictionId: ownProps.rootJurisdictionId,
  };
  const structuresCount = structureCountSelector(state, filters);
  const tree = treeSelector(state, filters);
  const jurisdictionsMetadata = getJurisdictionsMetadata(state);
  return { structuresCount, tree, jurisdictionsMetadata };
};

const mapDispatchToProps: MapDispatchToProps = {
  autoSelectCreator: autoSelectNodes,
  fetchJurisdictionsMetadataCreator: fetchJurisdictionsMetadata,
  fetchTreeCreator: fetchTree,
};

export const ConnectedJurisdictionSelectionsSlider = connect(
  mapStateToProps,
  mapDispatchToProps
)(JurisdictionSelectionsSlider);
