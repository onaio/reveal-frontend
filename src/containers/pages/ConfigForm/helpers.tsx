import React from 'react';
import Loading from '../../../components/page/Loading';
import { NoDataComponent } from '../../../components/Table/NoDataComponent';
import { OPENSRP_API_BASE_URL } from '../../../configs/env';
import { getAcessTokenOrRedirect, growl } from '../../../helpers/utils';
import { newGetPayloadOptions } from '../../../services/opensrp';

/** drill down props */
export const drillDownProps = {
  paginate: false,
  renderNullDataComponent: () => <NoDataComponent />,
};

export const defaultConfigProps = {
  LoadingComponent: <Loading />,
  accessToken: getAcessTokenOrRedirect,
  baseURL: OPENSRP_API_BASE_URL,
  customAlert: growl,
  getPayload: newGetPayloadOptions,
};
