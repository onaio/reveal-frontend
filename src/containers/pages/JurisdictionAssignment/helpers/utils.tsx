import { useState } from 'react';
import { displayError } from '../../../../helpers/errors';
import { TreeNode } from '../../../../store/ducks/opensrp/hierarchies/types';
import { nodeIsSelected } from '../../../../store/ducks/opensrp/hierarchies/utils';

/** tells whether we should select the parent checkbox in the table header
 * @param currentParentNode - the current parent node
 * @param currentChildren - current children
 */
export const checkParentCheckbox = (
  currentParentNode: TreeNode | undefined,
  currentChildren: TreeNode[]
) => {
  let selected = true;
  if (currentParentNode) {
    selected = selected && nodeIsSelected(currentParentNode);
  } else {
    // when currentParent node is undefined but there are currentChildren usually at the top level
    if (currentChildren.length > 0) {
      currentChildren.forEach(node => {
        selected = selected && nodeIsSelected(node);
      });
    } else {
      selected = false;
    }
  }
  return selected;
};

export const useHandleBrokenPage = () => {
  const [broken, setBroken] = useState<boolean>(false);
  const [errorMessage, setErrorMessage] = useState<string | undefined>();

  /** Convenience function to handle cases where we must abort and tell the user we have done so */
  const handleBrokenPage = (message: string) => {
    displayError(Error(message));
    setErrorMessage(message);
    setBroken(true);
  };

  return { broken, errorMessage, handleBrokenPage };
};
