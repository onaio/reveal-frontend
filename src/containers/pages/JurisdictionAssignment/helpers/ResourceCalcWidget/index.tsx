/** Resource Calculation component
 * going to be a side panel that calculates the ETA for an available
 * number of teams to be visit a variable number of structures in selected
 * jurisdictions.
 */

import reducerRegistry from '@onaio/redux-reducer-registry';
import { Field, Form, Formik } from 'formik';
import React from 'react';
import { connect } from 'react-redux';
import { Store } from 'redux';
import { format } from 'util';
import { ErrorPage } from '../../../../../components/page/ErrorPage';
import {
  AT_A_RATE_OF,
  NO_JURISDICTION_SELECTIONS_FOUND,
  RESOURCE_ESTIMATE_FOR,
  STRUCTURES_PER_TEAM_PER_DAY,
  TEAMS,
} from '../../../../../configs/lang';
import hierarchyReducer, {
  getParentNodeInSelectedTree,
  reducerName as hierarchyReducerName,
} from '../../../../../store/ducks/opensrp/hierarchies';
import { TreeNode } from '../../../../../store/ducks/opensrp/hierarchies/types';
import { getNodeStructureCount } from '../../../../../store/ducks/opensrp/hierarchies/utils';
import './index.css';

reducerRegistry.register(hierarchyReducerName, hierarchyReducer);

/** props */
export interface ResourceCalculationProps {
  rootId: string;
  planId: string;
  currentParentNode?: TreeNode;
  currentParentId?: string;
}

/** helper function that computes the estimate time to completion given a
 * the total structures and rate of completion per team
 * @param totalStructures - total structures in these jurisdiction
 * @param structuresPerTeam - rate of completion; how many structures a team can handler
 * @param teams - tne number of teams
 */
export const computeEstimate = (
  totalStructures: number,
  structuresPerTeam: number,
  teams: number
) => {
  if (!teams || !structuresPerTeam) {
    return 0;
  }
  return Math.ceil(totalStructures / (teams * structuresPerTeam));
};

/** The component that renders the resource calculation info */
const ResourceCalculation = (props: ResourceCalculationProps) => {
  const { currentParentNode } = props;

  if (!currentParentNode) {
    return <ErrorPage errorMessage={NO_JURISDICTION_SELECTIONS_FOUND} />;
  }
  const jurisdictionName = currentParentNode.model.label;
  const structuresCount = getNodeStructureCount(currentParentNode);

  return (
    <div id="resource-calc-widget">
      <h3 className="section-title">{format(RESOURCE_ESTIMATE_FOR, jurisdictionName)}</h3>

      <Formik
        initialValues={{ structuresCount: 0, teamsCount: 0 }}
        // tslint:disable-next-line: jsx-no-lambda
        onSubmit={() => {
          return;
        }}
      >
        {formikProps => {
          const { structuresCount: formikStructureSCount, teamsCount } = formikProps.values;
          const numDays = computeEstimate(structuresCount, formikStructureSCount, teamsCount);
          return (
            <Form>
              <p>
                {`${numDays} days`}
                <br />
                {AT_A_RATE_OF} <Field type="number" name="structuresCount" />
                {'  '}
                {STRUCTURES_PER_TEAM_PER_DAY} <Field type="number" name="teamsCount" /> {TEAMS}
              </p>
            </Form>
          );
        }}
      </Formik>
    </div>
  );
};

export { ResourceCalculation };

type MapStateToProps = Pick<ResourceCalculationProps, 'currentParentNode'>;

export const parentNodeSelector = getParentNodeInSelectedTree();

/** map props to store state selectors
 * @param state - the store
 * @param ownProps - component props
 */
const mapStateToProps = (
  state: Partial<Store>,
  ownProps: ResourceCalculationProps
): MapStateToProps => {
  const filters = {
    currentParentId: ownProps.currentParentId,
    planId: ownProps.planId,
    rootJurisdictionId: ownProps.rootId,
  };
  const currentParentNode = parentNodeSelector(state, filters);
  return { currentParentNode };
};

const ConnectedResourceWidget = connect(mapStateToProps)(ResourceCalculation);
export { ConnectedResourceWidget };
