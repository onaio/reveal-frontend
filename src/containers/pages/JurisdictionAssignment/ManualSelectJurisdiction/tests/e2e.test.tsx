import reducerRegistry from '@onaio/redux-reducer-registry';
import { mount } from 'enzyme';
import React from 'react';
import { act } from 'react-dom/test-utils';
import { Provider } from 'react-redux';
import { MemoryRouter, Route, Switch } from 'react-router';
import ConnectedJurisdictionAssignmentView from '..';
import { MANUAL_ASSIGN_JURISDICTIONS_URL } from '../../../../../constants';
import { renderTable } from '../../../../../helpers/testUtils';
import store from '../../../../../store';
import hierarchiesReducer, {
  reducerName as hierarchiesReducerName,
} from '../../../../../store/ducks/opensrp/hierarchies';
import { raZambiaHierarchy } from '../../../../../store/ducks/opensrp/hierarchies/tests/fixtures';
import plansReducer, { reducerName } from '../../../../../store/ducks/opensrp/PlanDefinition';
import { plans } from '../../../../../store/ducks/opensrp/PlanDefinition/tests/fixtures';

reducerRegistry.register(reducerName, plansReducer);
reducerRegistry.register(hierarchiesReducerName, hierarchiesReducer);

/* tslint:disable-next-line no-var-requires */
const fetch = require('jest-fetch-mock');

jest.mock('reactstrap', () => {
  const original = require.requireActual('reactstrap');
  return { ...original, Tooltip: () => null };
});

jest.mock('../../../AssigmentMapWrapper', () => {
  const mockComponent2 = (_: any) => <div id="mockComponent2">Assignment Wrapper</div>;
  return {
    ConnectedAssignmentMapWrapper: mockComponent2,
  };
});

it('plans with existing node selections', async () => {
  jest.setTimeout(10000);
  const envModule = require('../../../../../configs/env');
  envModule.ASSIGNMENT_PAGE_SHOW_MAP = true;

  const App = () => {
    return (
      <Switch>
        <Route
          exact={true}
          path={`${MANUAL_ASSIGN_JURISDICTIONS_URL}/:planId/:rootId/:parentId`}
          component={ConnectedJurisdictionAssignmentView}
        />
        <Route
          exact={true}
          path={`${MANUAL_ASSIGN_JURISDICTIONS_URL}/:planId/:rootId`}
          component={ConnectedJurisdictionAssignmentView}
        />
      </Switch>
    );
  };

  const plan = plans[0];
  // plan should have selections: Akros_3, ra_CDZ_139e
  plan.jurisdiction = [
    { code: '45017166-cc14-4f2f-b83f-4a72ce17bf91' },
    { code: 'fa351dc7-478c-4be9-be60-de05827dba8e' },
  ];
  const rootId = '0ddd9ad1-452b-4825-a92a-49cb9fc82d18';
  fetch
    .once(JSON.stringify(plan), { status: 200 })
    .once(JSON.stringify(raZambiaHierarchy), { status: 200 });

  const wrapper = mount(
    <Provider store={store}>
      <MemoryRouter
        initialEntries={[
          {
            hash: '',
            pathname: `${MANUAL_ASSIGN_JURISDICTIONS_URL}/${plan.identifier}/${rootId}`,
            search: '',
            state: {},
          },
        ]}
      >
        <App />
      </MemoryRouter>
    </Provider>
  );

  await act(async () => {
    await new Promise(resolve => setImmediate(resolve));
    wrapper.update();
  });

  // check props given to mock component
  const passedProps: any = wrapper.find('mockComponent2').props();
  expect(passedProps.currentParentId).toBeUndefined();
  expect(passedProps.rootJurisdictionId).toEqual(rootId);

  // rendered component
  expect(wrapper.text()).toMatchSnapshot('general snapshot of page');

  // we now need to check that existing assignments are selected
  renderTable(wrapper, 'should have single node(parent node)');
  let tbodyRow = wrapper.find('tbody tr');
  expect(tbodyRow.length).toEqual(1);
  expect(
    tbodyRow
      .at(0)
      .text()
      .includes('Zambia')
  ).toBeTruthy();

  // drilldown to Akros_3 and check if its selected
  // drilldown to ra Zambia
  tbodyRow
    .at(0)
    .find('NodeCell Link a')
    .simulate('click', { button: 0 });
  await act(async () => {
    await new Promise(resolve => setImmediate(resolve));
    wrapper.update();
  });
  renderTable(wrapper, 'after first click');
  // nodes are not selected because none of them is akros_3
  wrapper.find('tbody tr').forEach(tr =>
    expect(
      tr
        .find('td input')
        .at(0)
        .props().checked
    ).toBeFalsy()
  );

  // drillDown to Lusaka
  tbodyRow = wrapper.find('tbody tr');
  tbodyRow
    .at(0)
    .find('NodeCell Link a')
    .simulate('click', { button: 0 });
  await act(async () => {
    await new Promise(resolve => setImmediate(resolve));
    wrapper.update();
  });
  renderTable(wrapper, 'after second click');
  // nodes are not selected because none of them is akros_3
  expect(tbodyRow.length).toEqual(4);
  expect(
    tbodyRow
      .at(0)
      .text()
      .includes('Mtendere')
  ).toBeTruthy();
  wrapper.find('tbody tr').forEach(tr =>
    expect(
      tr
        .find('td input')
        .at(0)
        .props().checked
    ).toBeFalsy()
  );

  // drillDown to Mtendere where akros_3 lies
  tbodyRow = wrapper.find('tbody tr');
  tbodyRow
    .at(0)
    .find('NodeCell Link a')
    .simulate('click', { button: 0 });
  await act(async () => {
    await new Promise(resolve => setImmediate(resolve));
    wrapper.update();
  });
  renderTable(wrapper, 'after third click');
  // akros_3 node should  be selected while the others are not
  expect(tbodyRow.length).toEqual(2);
  wrapper.find('tbody tr').forEach(tr => {
    if (tr.text().includes('Akros_3')) {
      expect(
        tr
          .find('td input')
          .at(0)
          .props().checked
      ).toBeTruthy();
    } else {
      expect(
        tr
          .find('td input')
          .at(0)
          .props().checked
      ).toBeFalsy();
    }
  });
});
