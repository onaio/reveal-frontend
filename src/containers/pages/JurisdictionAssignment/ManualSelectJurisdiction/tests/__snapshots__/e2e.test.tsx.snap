// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`plans with existing node selections: after first click 1`] = `"NameStructures CountTarget StatusStatus setting by...Selected Jurisdictions"`;

exports[`plans with existing node selections: after first click 2`] = `"Lusaka131"`;

exports[`plans with existing node selections: after first click 3`] = `"ra Eastern01"`;

exports[`plans with existing node selections: after first click 4`] = `"ra Luapula00"`;

exports[`plans with existing node selections: after first click 5`] = `"ra Lusaka00"`;

exports[`plans with existing node selections: after second click 1`] = `"NameStructures CountTarget StatusStatus setting by...Selected Jurisdictions"`;

exports[`plans with existing node selections: after second click 2`] = `"Mtendere131"`;

exports[`plans with existing node selections: after second click 3`] = `"Lusaka District0Not Targeted0"`;

exports[`plans with existing node selections: after third click 1`] = `"NameStructures CountTarget StatusStatus setting by...Selected Jurisdictions"`;

exports[`plans with existing node selections: after third click 2`] = `"REVEAL TEST0Not Targeted0"`;

exports[`plans with existing node selections: after third click 3`] = `"Akros_113Not Targeted0"`;

exports[`plans with existing node selections: after third click 4`] = `"Akros_20Not Targeted0"`;

exports[`plans with existing node selections: after third click 5`] = `"Akros_30TargetedUser Change1"`;

exports[`plans with existing node selections: general snapshot of page 1`] = `"Planning toolA2-Lusaka Akros Test Focus 2Assign JurisdictionsAssignment Wrapper....NameStructures CountTarget StatusStatus setting by...Selected Jurisdictionsra Zambia132save draftSave & Activate"`;

exports[`plans with existing node selections: should have single node(parent node) 1`] = `"NameStructures CountTarget StatusStatus setting by...Selected Jurisdictions"`;

exports[`plans with existing node selections: should have single node(parent node) 2`] = `"ra Zambia132"`;
